frontend/heart-code/src/components/auth/AuthValidation.ts

```diff
@@ -2,4 +2,35 @@ function isValidEmail(email: string): boolean {
   return new RegExp("^\\S+@\\S+\\.\\S+$").test(email);
 }
 
-export { isValidEmail };
+function isValidPassword(password: string): boolean {
+  if (password == null || password.length < 7) {
+    return false;
+  } else {
+    let numOfUppercase = 0;
+    let numOfLowercase = 0;
+    let numOfDigits = 0;
+    let numOfSpecialChars = 0;
+    for (let i = 0; i < password.length; i++) {
+      let ch = password.charAt(i);
+      let format = /[ `!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~]/;
+      if (format.test(ch)) {
+        numOfSpecialChars++;
+      } else if (/^\d+$/.test(ch)) {
+        numOfDigits++;
+      } else if (ch === ch.toUpperCase()) {
+        numOfUppercase++;
+      } else if (ch === ch.toLowerCase()) {
+        numOfLowercase++;
+      }
+    }
+
+    return (
+      numOfUppercase >= 1 &&
+      numOfLowercase >= 1 &&
+      numOfDigits >= 1 &&
+      numOfSpecialChars >= 1
+    );
+  }
+}
```

Comment:
in de frontend is het wel van belang dat de user snapt _waarom_ het wachtwoord niet goed is, hier zou ik dus per requirement een error message sturen, bijvoorbeeld door het return-type te veranderen van een boolean naar een object:
`
{
  isValid: boolean;
  errors: string[]
}
`
dan kun je onder de input fields aangeven wat er met het wachtwoord mis is

Reply:
misschien moet deze method dan voor de duidelijkheid hernoemt worden naar iets in de richting van 'validatePassword'

Reply:
Hij geeft nu inderdaad alleen een error die zegt wat er precies in moet, maar niet wat er mist. Ik dacht zelf dat dat wel voldoende was, aangezien je het wachtwoord kan bekijken.
Maar ik zou inderdaad nog even per ding aan kunnen geven wat er mis is.
Met de methode naam bedoel je die op de backend of de frontend?
Wel even pauze eerst :d

Reply:
ik zie inderdaad dat er op een andere plek een de algemene password requirements worden aangegeven, dus dat is wel goed zo inderdaad